Fix error connection behaviour

Sych474 opened the pull request:
Fixes # Fix error connection behaviour. 
Changes:
- Set State = Broken to the connection if catch an error
- Some code-style fixes

Sych474 committed:
Add to connection logic to set it broken on error.

Sych474 committed:
Use new connection logic to fix errors.

ikopylov commented on ClickHouse.Ado/ClickHouseCommand.cs:
```diff
@@ -18,15 +18,13 @@ public class ClickHouseCommand
         : IDbCommand
 #endif
     {
-        private ClickHouseConnection _clickHouseConnection;
```
Can you revert this change? It can be problematic to merge changes from original repository in future due to conflicts

ikopylov commented on ClickHouse.Ado/ClickHouseConnection.cs:
```diff
@@ -33,8 +33,7 @@ public ClickHouseConnection(string connectionString)
         private BinaryWriter _writer;*/
-        internal ProtocolFormatter Formatter { get;
-            set; }
+        internal ProtocolFormatter Formatter { get; set; }
```
All formatting changes should be reverted due to high possibility of merge conflicts in future

Sych474 committed:
start reverting code-style changes

fix revert code-style

revert code-style

revert code-style

fix

fix

fix

revert code-style


revert code-style
